backend/app/main.py

```diff
@@ -131,6 +132,23 @@ async def update_document(
     return db_document
 
 
+@app.patch("/api/erase_assignments")
+async def erase_assignments(
+    data: AssignmentsCreate, session: Session = Depends(get_session)
+):
```

Comment:
H: Did you try using the existing `api/update_assignments` endpoint, passing nulls for the `zone`? We'll have to allow `zone` to be nullable (and possible run a migration).

```python
class AssignmentsBase(SQLModel):
    metadata = MetaData(schema=DOCUMENT_SCHEMA)
    document_id: str = Field(sa_column=Column(UUIDType, primary_key=True))
    geo_id: str = Field(primary_key=True)
    zone: int | None # <- right here!
```

If we can get away with a single endpoint I'd prefer that.

If we decide we prefer to have two separate endpoints then I would recommend we reimplement this w/o the slow python loop. Pydantic 2.0 was reimplemented in rust, which should give us really good perf when dumping the model (allowing for large document updates at a time). I think the python loop implementation will result in longer running updates, but that's just a guess.

Reply:
Sounds good, I'll create a migration around NOT NULL